modules/gapi/test/infer/gapi_infer_onnx_test.cpp

```diff
@@ -262,6 +332,54 @@ class ONNXGRayScaleTest : public ONNXtest {
         dst = dst.reshape(1, {1, 1, new_h, new_w});
     }
 };
+
+class ONNXWithRemap : public ONNXtest {
+public:
+    // You can specify any size of the outputs, since we don't know infer result
+    // Tests validate a range with results and don't compare empty space
+    void validate() {
+        GAPI_Assert(!out_gapi.empty() && !out_onnx.empty());
+        ASSERT_EQ(out_gapi.size(), out_onnx.size());
+        const auto size = out_onnx.size();
+        for (size_t i = 0; i < size; ++i) {
+            float* op = out_onnx.at(i).ptr<float>();
+            float* gp = out_gapi.at(i).ptr<float>();
+            const auto out_size = std::min(out_onnx.at(i).total(), out_gapi.at(i).total());
+            for (size_t d_idx = 0; d_idx < out_size; ++d_idx) {
```

Comment:
These checks are not reliable.
If `out_gapi.at(i).total()` is zero then no checks are performed at all.

Reply:
Added assert for zero size check.